src/cdk/table/table.ts

```diff
@@ -979,9 +1025,12 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       throw getTableUnknownDataSourceError();
     }
 
-    this._renderChangeSubscription = dataStream!.pipe(takeUntil(this._onDestroy))
-      .subscribe(data => {
-        this._data = data || [];
+    this._renderChangeSubscription = combineLatest([dataStream!, this.viewChange]).pipe(
+        takeUntil(this._onDestroy))
+      .subscribe(([data, range]) => {
+        this._data = data;
+        this._renderedRange = range;
```

Comment:
Nit: Should this be `_renderRange` or `_dataRange` to imply that this is the range we intend render, rather than the one we have rendered?